vendor/mlir/mlir.BUILD

```diff
@@ -665,3 +665,24 @@ cc_library(
     ],
     alwayslink = 1,
 )
+
+cc_library(
+    name = "MlirOptLib",
+    srcs = glob([
+        "lib/Support/MlirOptMain.cpp",
+    ]),
+    hdrs = [
+        "include/mlir/Support/MlirOptMain.h",
+    ],
+    copts = PLATFORM_COPTS,
+    includes = ["include"],
+    deps = [
+        ":Analysis",
+        ":IR",
+        ":Parser",
+        ":Pass",
+        ":Support",
+        ":Transforms",
+        "@llvm//:support",
+    ],
+)
```

Comment:
There's already an `OptMain` cc_library defined, is this different somehow?

Reply:
If we switch to using `MlirOptLib` that's fine, but we'll need to remove the old `OptMain` if that's the case. Or we just stick with `OptMain`, doesn't matter much to me.